test/system/lifecycle/lifecycle_test.go

```diff
@@ -0,0 +1,281 @@
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/gardener/gardener-extension-shoot-oidc-service/pkg/constants"
+	"github.com/gardener/gardener-extension-shoot-oidc-service/test/system/resources/templates"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gutil "github.com/gardener/gardener/pkg/utils/gardener"
+	"github.com/gardener/gardener/test/framework"
+
+	appsv1 "k8s.io/api/apps/v1"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func init() {
+	framework.RegisterShootFrameworkFlags()
+}
+
+const (
+	timeout = 30 * time.Minute
+)
+
+var _ = Describe("Shoot oidc service testing", func() {
+	f := framework.NewShootFramework(nil)
+	f.ResourcesDir = "../resources"
+
+	f.Serial().Release().CIt("Should perform the common case scenario without any errors", func(ctx context.Context) {
+		err := f.UpdateShoot(ctx, func(shoot *gardencorev1beta1.Shoot) error {
+			if !isOIDCServiceEnabled(shoot) {
+				shoot.Spec.Extensions = append(shoot.Spec.Extensions, gardencorev1beta1.Extension{
+					Type: constants.ExtensionType,
+				})
+			}
+			return nil
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		oidcDeployment := &appsv1.Deployment{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.ApplicationName,
+				Namespace: f.ShootSeedNamespace(),
+			},
+		}
+
+		// Verify that the oidc deployment exists and is deployed with the correct number of replicas
+		err = f.SeedClient.Client().Get(ctx, client.ObjectKeyFromObject(oidcDeployment), oidcDeployment)
+		Expect(err).ToNot(HaveOccurred())
+		one := int32(1)
+		Expect(*oidcDeployment.Spec.Replicas).To(Equal(one))
+		Expect(oidcDeployment.Status.ReadyReplicas).To(Equal(one))
+
+		oidConfig, err := getOIDConfig(ctx, f.SeedClient.RESTClient())
+		Expect(err).ToNot(HaveOccurred())
+
+		jwksURL, err := url.Parse(oidConfig.JWKSURL)
+		Expect(err).ToNot(HaveOccurred())
+
+		jwks, err := getJWKS(ctx, f.SeedClient.RESTClient(), jwksURL.Path)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Deploy oidc resource
+		clientId := "some-custom-client-id"
+		values := map[string]interface{}{
+			"issuerURL": oidConfig.Issuer,
+			"clientID":  clientId,
+			"keys":      base64.StdEncoding.EncodeToString(jwks),
+		}
+		err = f.RenderAndDeployTemplate(ctx, f.ShootClient, templates.CustomOIDCConfiguration, values)
+		Expect(err).ToNot(HaveOccurred())
+		getRequest := f.ShootClient.RESTClient().Get()
+		getRequest = getRequest.RequestURI("/apis/authentication.gardener.cloud/v1alpha1/openidconnects/custom")
+		getResponse := getRequest.Do(ctx)
+		Expect(getResponse.Error()).ToNot(HaveOccurred())
```

Comment:
Can you use the controller runtime client `f.ShootClient.Client()` which dynamically discover resources?
It should be something like 
```golang 
oidc := openidconnectv1alpha1.OpenIDConnect{} // openidconnectv1alpha1 "github.com/gardener/oidc-webhook-authenticator/apis/authentication/v1alpha1"
err = f.ShootClient.Client().Get(ctx, types.NamespacedName{Name: "custom"}, &oidc)
Expect(err).ToNot(HaveOccurred())
```